app_feup/lib/model/entities/Course.dart

```diff
@@ -0,0 +1,48 @@
+class Course {
+  int id;
+  int fest_id;
+  String name;
+  String abbreviation;
+  String currYear;
+  int firstEnrollment;
+  String state = "";
+
+  Course.secConstructor(int id, int fest_id, String name, String abbreviation, String currYear, int firstEnrollment, String state) {
+    this.id = id;
+    this.fest_id = fest_id;
+    this.name = name;
+    this.abbreviation = abbreviation;
+    this.currYear = currYear;
+    this.firstEnrollment = firstEnrollment;
+    this.state = state;
+  }
+
+  Course(
+      {int this.id,
+        int this.fest_id,
+        String this.name,
+        String this.abbreviation,
+        String this.currYear,
+        int this.firstEnrollment});
+
```

Comment:
Is there a need to create a method just to have a second constructor? Dart does support default values for attributes

Reply:
But named constructors are cooool






(usei default values)